internal/services/sources_service.go

```diff
@@ -16,7 +16,11 @@ func ListSources(w http.ResponseWriter, r *http.Request) {
 		renderError(w, r, payloads.NewClientInitializationError(r.Context(), "sources client", err))
 		return
 	}
-	resp, err := client.ListSourcesWithResponse(r.Context(), &sources.ListSourcesParams{}, AddIdentityHeader)
+
+	// name := chi.URLParamFromCtx(r.Context(), "source_type")
+	filter := sources.QueryFilter("[source_type][name]=amazon")
+	resp, err := client.ListSourcesWithResponse(r.Context(), &sources.ListSourcesParams{Filter: &filter}, AddIdentityHeader)
+
```

Comment:
From what I understood this is the correct way in order to add filters to their request but it does not work for me. 
When I am requesting sources directly, it works: 
curl --location -g  --request GET 'http://localhost:8000/api/sources/v3.1/sources?filter[source_type][name]=amazon' --header 'x-rh-identity: IDENTITY'
but when I am trying it through our service this is the query that is being formed using: https://github.com/RHEnVision/provisioning-backend/blob/eeb276f93a8629d114b2a89e72ea8c978e6bc3bc/internal/clients/sources/client.gen.go#L1229
```golang
http://localhost:8000/api/sources/v3.1/sources?filter=%3F%5Bsource_type%5D%5Bname%5D%3Damazon
``` 
and this is the error I am getting on sources side :
```golang
runtime error: index out of range [1] with length 1 goroutine 106 
```
@lzap, @ezr-ondrej Can you take a look as well?